modules/system/assets/js/snowboard/main/Snowcart.js

```diff
@@ -0,0 +1,347 @@
+import PluginBase from '../abstracts/PluginBase';
+import Singleton from '../abstracts/Singleton';
+import PluginLoader from './PluginLoader';
+
+import Debounce from '../utilities/Debounce';
+import JsonParser from '../utilities/JsonParser';
+import Sanitizer from '../utilities/Sanitizer';
+
+/**
+ * Snowboard - the Winter JavaScript framework.
+ *
+ * This class represents the base of a modern take on the Winter JS framework, being fully extensible and taking advantage
+ * of modern JavaScript features by leveraging the Laravel Mix compilation framework. It also is coded up to remove the
+ * dependency of jQuery.
+ *
+ * @copyright 2021 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ * @link https://wintercms.com/docs/snowboard/introduction
+ */
+export default class Snowboard {
+    /**
+     * Constructor.
+     *
+     * @param {boolean} autoSingletons Automatically load singletons when DOM is ready. Default: `true`.
+     * @param {boolean} debug Whether debugging logs should be shown. Default: `false`.
+     */
+    constructor(autoSingletons, debug) {
+        /* develblock:start */
+        this.debugEnabled = (typeof debug === 'boolean' && debug === true) ? true : false;
+        /* develblock:end */
```

Comment:
The original intention was that when Snowboard was compiled for production, it would strip out the debugging method and calls to bring the filesize down, since there's no need for debugging in production. However, given that we're going to include debugging but only enable it when `app.debug` is true, I'm going to remove these comments.